### PR TITLE
Feature/sckan 340 - Adjust default Sentence List view properties

### PR DESCRIPTION
**Pull request by D-GopalKrishna**
Jira link - [SCKAN-340](https://metacell.atlassian.net/browse/SCKAN-340)

**Task description:**

Sorting by last edited is straight forward. Sorting by owner is done as following rule (please let me know if this sounds wrong or any other feedback):

- keep the empty owner Sentences last; at all times. Reasoning behind it - if you are trying to sort by owner, then you should see the most relevant information first (which is not the empty owner). So for both ascending and descending sorting - we keep the empty owners at last.
- If no sorting is applied - you see - Sentences sorted by - date (second priority) and owner (first priority) (see owner’s Sentences first)
- If sorting by last edited - asc (see oldest first) or desc (see latest first)
- If sorting by owner:
    - asc - keep the empty owner Sentences last (first priority) → See the owner’s Sentences first (second priority) → Sort by owner’s full name in asc order (third priority)
    - desc - keep the empty owner Sentences last (first priority) → See the owner’s Sentences last (second priority) → Sort by owner’s full name in desc order (third priority)

[SCKAN-340]: https://metacell.atlassian.net/browse/SCKAN-340?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by afonsobspinto — CHANGES_REQUESTED** [review]
I believe this point is missing from the acceptance criteria 
>3. Apply filters by default upon login to show only sentences with the following statuses: compose later; compose now; needs further review; open; ready to compose.

**Comment by D-GopalKrishna** [comment]
> I believe this point is missing from the acceptance criteria
> 
> > 3. Apply filters by default upon login to show only sentences with the following statuses: compose later; compose now; needs further review; open; ready to compose.


thanks alot @afonsobspinto ! here fixed. 
image

**Review by afonsobspinto — DISMISSED** [review]

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -34,18 +36,23 @@ function App() {
   if (userProfile.isSignedIn()) {
     const user = userProfile.getProfile();
 
-    const defaultStateFilter = []
+    const defaultStatementStateFilter = []
+    const defaultSentenceStateFilter = [
+      ComposerSentenceListStateEnum.ComposeLater, ComposerSentenceListStateEnum.ComposeNow,
```
Comment (ddelpiano) [inline frontend/src/App.tsx:41]: Missing the open state.

From the acceptance criteria:
Apply filters by default upon login to show only sentences with the following statuses: compose later; compose now; needs further review; **open**; ready to compose.

**Review by ddelpiano — CHANGES_REQUESTED** [review]

**Comment by ddelpiano** [comment]
@D-GopalKrishna a bonus request, could we change the last edit for a sentence or a statement if we add/remove a tag from this record or there is a new message? Thanks

**Comment by D-GopalKrishna** [comment]
> @D-GopalKrishna a bonus request, could we change the last edit for a sentence or a statement if we add/remove a tag from this record or there is a new message? Thanks

done @ddelpiano, thanks!! 
here's a v-demo 

https://github.com/user-attachments/assets/eec5b320-870a-4a05-b016-244c2b83c117

**Review by ddelpiano — APPROVED** [review]

**Pull request merged**